persistence/src/commonMain/kotlin/com/wire/kalium/persistence/db/DatabaseOptimizer.kt

```diff
@@ -0,0 +1,37 @@
+/*
+ * Wire
+ * Copyright (C) 2025 Wire Swiss GmbH
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see http://www.gnu.org/licenses/.
+ */
+package com.wire.kalium.persistence.db
+
+import com.wire.kalium.util.DelicateKaliumApi
+import com.wire.kalium.util.KaliumDispatcher
+import com.wire.kalium.util.KaliumDispatcherImpl
+import kotlinx.coroutines.withContext
+
+@DelicateKaliumApi("This class is used for debugging purposes only")
+class DatabaseOptimizer(
+    private val localDatabase: UserDatabaseBuilder,
+    private val dispatcher: KaliumDispatcher = KaliumDispatcherImpl
+) {
+    suspend fun optimize() = withContext(dispatcher.io) {
```

Comment:
Here are a couple of pragma configs that I was playing sometime ago, while playing with Infinite monkeys that could potentially help us in squeezing more perf.

https://github.com/wireapp/kalium/compare/develop...chore/tune-sqlite#diff-89d9522d3a6edc443a59ea0bce3b1008cb25cfffd7513ca7820ada5ba99a3c38R356-R377


https://www.sqlite.org/pragma.html#pragma_synchronous
https://www.sqlite.org/tempfiles.html#transient_indices
https://www.sqlite.org/pragma.html#pragma_incremental_vacuum

Reply:
Thanks for ideas! We can iterate on it. If this PR does not solve performance issue we can tackle this right away 👍